main.go

```diff
@@ -67,6 +74,8 @@ func main() {
 			line, err := in.ReadSlice('\n')
 			must(err)
 			//log.Printf("[IN]  %s", line)
+			// If we've read something, feed the net.Dial Deadline watchdog
+			feedWatchdog(c)
```

Comment:
I believe it is functionally equivalent to call `c.SetReadDeadline(...)` the line before `in.ReadSlice`, and would remove the duplication (and therefore need for a dedicated function)